de.metas.business/src/main/java/org/adempiere/impexp/DiscountSchemaImportProcess.java

```diff
@@ -0,0 +1,224 @@
+	{
+		return new X_I_DiscountSchema(ctx, rs, ITrx.TRXNAME_ThreadInherited);
+	}
+
+	@Override
+	protected ImportRecordResult importRecord(@NonNull final IMutable<Object> state, @NonNull final I_I_DiscountSchema importRecord) throws Exception
+	{
+		//
+		// Get previous values
+		MDiscountSchemaImportContext context = (MDiscountSchemaImportContext)state.getValue();
+		if (context == null)
+		{
+			context = new MDiscountSchemaImportContext();
+			state.setValue(context);
+		}
+		final I_I_DiscountSchema previousImportRecord = context.getPreviousImportRecord();
+		final int previousDiscountSchemaId = context.getPreviousM_DiscountSchema_ID();
+		final String previousBPartnerValue = context.getPreviousBPartnerValue();
+		context.setPreviousImportRecord(importRecord);
+
+		final ImportRecordResult schemaImportResult;
+
+		final boolean firstImportRecordOrNewDiscountSchema = previousImportRecord == null
+				|| !Objects.equals(importRecord.getBPartner_Value(), previousBPartnerValue);
+
+		if (firstImportRecordOrNewDiscountSchema)
+		{
+			// create a new list because we are passing to a new discount
+			context.clearPreviousRecordsForSameDiscountSchema();
+			schemaImportResult = importDiscountSchema(importRecord);
+		}
+		else
+		{
+			if (previousDiscountSchemaId <= 0)
+			{
+				schemaImportResult = importDiscountSchema(importRecord);
+			}
+			else if (importRecord.getM_DiscountSchema_ID() <= 0 || importRecord.getM_DiscountSchema_ID() == previousDiscountSchemaId)
+			{
+				schemaImportResult = doNothingAndUsePreviousDiscountSchema(importRecord, previousImportRecord);
+			}
+			else
+			{
+				throw new AdempiereException("Same value or movement date as previous line but not same Inventory linked");
+			}
+		}
+
+		importDiscountSchemaBreak(importRecord);
+		context.collectImportRecordForSameDiscountSchema(importRecord);
+
+		return schemaImportResult;
+	}
+
+	private ImportRecordResult importDiscountSchema(@NonNull final I_I_DiscountSchema importRecord)
+	{
+		final ImportRecordResult inventoryImportResult;
+		inventoryImportResult = importRecord.getM_DiscountSchema_ID() <= 0 ? ImportRecordResult.Inserted : ImportRecordResult.Updated;
+
+		final I_M_DiscountSchema discountSchema;
+		if (importRecord.getM_DiscountSchema_ID() <= 0)
+		{
+			discountSchema = createNewMDiscountSchemas(importRecord);
+			final I_C_BPartner bpartner = importRecord.getC_BPartner();
+			bpartner.setM_DiscountSchema(discountSchema);
+			InterfaceWrapperHelper.save(bpartner);
+		}
+		else
+		{
+			discountSchema = importRecord.getM_DiscountSchema();
+		}
+
+		ModelValidationEngine.get().fireImportValidate(this, importRecord, discountSchema, IImportInterceptor.TIMING_AFTER_IMPORT);
+		InterfaceWrapperHelper.save(discountSchema);
+		importRecord.setM_DiscountSchema_ID(discountSchema.getM_DiscountSchema_ID());
+		return inventoryImportResult;
+	}
+
+	private I_M_DiscountSchema createNewMDiscountSchemas(I_I_DiscountSchema importRecord)
```

Comment:
scnr... `createNewMDiscountSchemas`  ... without the trailing `s` because only one schema is created.
You can even lose that `M` because makes reading harder.
So you could rename it to something like `createNewDiscountSchema`